### PR TITLE
Unified world mocks

### PR DESCRIPTION
**Pull request by andrei-marinica**
Unified the following blockchain mocks into one:
- arwen-mandos
- arwen-debug
- arwen context tests

Also moved the mandos parser to a more visible place.

### CODE REVIEW

**Comment by camilbancioiu** [comment]
Optional: please rename `mock/context` to `mock/contexts`, to match the name of `arwen/contexts`.

**Review by andreibancioiu — DISMISSED** [review]

File: mock/world/worldDef.go [hunk 1]
```diff
@@ -23,6 +24,10 @@ type BlockchainHookMock struct {
 	Blockhashes                  [][]byte
 	mockAddressGenerationEnabled bool
 	NewAddressMocks              []*NewAddressMock
+	StateRootHash                []byte
+	Err                          error
+	LastCreatedContractAddress   []byte
+	CompiledCode                 map[string][]byte
```
Comment (camilbancioiu) [inline mock/world/worldDef.go:30]: Suggest to rename this struct to `MockWorld`.

File: mock/world/worldUtil.go [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package callbackblockchain
```
Comment (camilbancioiu) [inline mock/world/worldUtil.go:1]: `package mockworld`

File: mock/world/worldDef.go [hunk 3]
```diff
@@ -13,6 +13,7 @@ type BlockInfo struct {
 	BlockNonce     uint64
```
[the remark this answers is not in the document]
Comment (camilbancioiu) [reply mock/world/worldDef.go:13]: Or `worldmock`.

**Review by andreibancioiu — APPROVED** [review]

**Review by camilbancioiu — APPROVED** [review]

File: mock/world/worldCallbacks.go [hunk 4]
```diff
@@ -16,26 +16,23 @@ var zero = big.NewInt(0)
...
-
-		result[29] = byte(creatorNonce)
-
-		copy(result[30:], creatorAddress[30:])
-
+		result := GenerateMockAddress(creatorAddress, creatorNonce)
+		b.LastCreatedContractAddress = result
 		return result, nil
 	}
 	// empty byte array signals not implemented, fallback to default
```
[the remark this answers is not in the document]
Comment (andrei-marinica) [reply mock/world/worldCallbacks.go]: you're right, I think it is a leftover from IELE days
I removed the mockAddressGenerationEnabled flag altogether, it never seems to be required as false, which gets rid of the if and of that extra empty result. That never happens IRL anyway.

**Review by andreibancioiu — APPROVED** [review]

File: arwen/host/execution_test.go [hunk 5]
```diff
@@ -27,7 +27,8 @@ const (
 )
 
 func TestNewArwen(t *testing.T) {
-	host, err := DefaultTestArwen(t, &contextmock.BlockchainHookStub{})
+	// host, err := DefaultTestArwen(t, &contextmock.BlockchainHookStub{})
```
Comment (camilbancioiu) [inline arwen/host/execution_test.go:30]: Delete commented line.

**Review by camilbancioiu — APPROVED** [review]

**Pull request merged**